daemons/fenced/fenced_commands.c

```diff
@@ -900,11 +900,17 @@ xml2device_params(const char *name, xmlNode *dev)
 static stonith_device_t *
 build_device_from_xml(xmlNode * msg)
 {
-    const char *value = NULL;
+    const char *value;
     xmlNode *dev = get_xpath_object("//" F_STONITH_DEVICE, msg, LOG_ERR);
     stonith_device_t *device = NULL;
+    char *agent = crm_element_value_copy(dev, "agent");
```

Comment:
Did you intend to use this variable somewhere?

Reply:
Stellar spot, responsibility for allocation was indeed meant to be
terminated in a subsequent assignment to `device` (replacing the
existing copy-to-heap), something that accidentaly stayed omitted.
Fixed now.